src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexThresholds.cs

```diff
@@ -22,19 +22,24 @@ internal static class SymbolicRegexThresholds
         /// processing starts out in DFA mode, even if we've previously triggered NFA mode for the same regex.
         /// We switch over into NFA mode the first time a given traversal (match operation) results in us needing
         /// to create a new node and the graph is already or newly beyond this threshold.
+        /// This limit is chosen due to memory usage constraints, the worst possible allocation is currently approx. 50 MB;
+        /// There is some motivation to make this configurable, as it can exchange upfront costs with potentially
+        /// significant search-time performance gains.
         /// </remarks>
-        internal const int NfaThreshold = 10_000;
+        internal const int NfaThreshold = 25_000;
 
         /// <summary>
         /// Default maximum estimated safe expansion size of a <see cref="SymbolicRegexNode{TSet}"/> AST
-        /// after the AST has been anlayzed for safe handling.
+        /// after the AST has been analyzed for safe handling.
         /// <remarks>
         /// If the AST exceeds this threshold then <see cref="NotSupportedException"/> is thrown.
         /// This default value may be overridden with the AppContext data
         /// whose name is given by  <see cref="SymbolicRegexSafeSizeThreshold_ConfigKeyName"/>.
         /// </remarks>
+        /// This limit is chosen due to worst case NFA speed constraints, which is about 150kb/s,
+        /// although it could be safely raised higher at the expense of worst-case NFA performance
         /// </summary>
-        internal const int DefaultSymbolicRegexSafeSizeThreshold = 1000;
+        internal const int DefaultSymbolicRegexSafeSizeThreshold = 10_000;
```

Comment:
We looked at all the regex patterns from RealWorldRegexes.json and the largest regexes there reached a bit over 5000 in AST size, so 10_000 was picked with some headroom. This should cover almost all large regex use cases

Reply:
Ok.